cmasher/tests/test_utils.py

```diff
@@ -65,6 +66,113 @@ def _MPL38_colormap_eq(cmap, other) -> bool:
 
 
 # %% PYTEST CLASSES AND FUNCTIONS
+# Pytest class for combine_cmaps
+class Test_combine_cmaps:
+    # Test if multiple Colormaps or colormap names can be combined
+    @pytest.mark.parametrize(
+        "cmaps, nodes",
+        [
+            (["Blues", "Oranges", "Greens"], [0.25, 0.75]),
+            (["Blues", "Oranges", "Greens"], np.array([0.25, 0.75])),
+            (
+                [
+                    mpl.colormaps["Blues"],
+                    mpl.colormaps["Oranges"],
+                    mpl.colormaps["Greens"],
+                ],
+                [0.25, 0.75],
+            ),
+        ],
+    )
+    def test_combine_cmaps(self, cmaps, nodes):
+        combined_cmap = combine_cmaps(*cmaps, nodes=nodes, n_rgb_levels=256)
+        blues_cmap = mpl.colormaps["Blues"]
+        oranges_cmap = mpl.colormaps["Oranges"]
+        greens_cmap = mpl.colormaps["Greens"]
+
+        assert np.allclose(combined_cmap(0.0), blues_cmap(0))
+        assert np.allclose(combined_cmap(0.25), oranges_cmap(0))
+        assert np.allclose(combined_cmap(0.75), greens_cmap(0))
+        assert np.allclose(combined_cmap(1.0), greens_cmap(255))
+
+        assert combined_cmap.N == 256
+
+    # Test if combining less than 2 colormaps triggers an error
+    @pytest.mark.parametrize(
+        "cmaps",
+        [
+            pytest.param([], id="no_cmap"),
+            pytest.param(["Blues"], id="single_cmap"),
+            pytest.param(["fake_name"], id="fake_cmap_name"),
+        ],
+    )
+    def test_not_enough_cmaps(self, cmaps):
+        with pytest.raises(
+            ValueError, match="Expected at least two colormaps to combine."
+        ):
+            combine_cmaps(*cmaps)
+
+    # Test if invalid colormap name raise an error
+    def test_invalid_cmap_name(self):
+        with pytest.raises(
+            KeyError,
+            match="'fake_cmap' is not a known colormap name",
+        ):
+            combine_cmaps("fake_cmap", "Blues")
+
+    # Test if invalid colormap types raise an error
+    @pytest.mark.parametrize(
+        "invalid_cmap",
+        [0, 0.0, [], ()],
+    )
+    def test_invalid_cmap_types(self, invalid_cmap):
+        with pytest.raises(
+            TypeError,
+            match=f"Unsupported colormap type: {type(invalid_cmap)}.",
+        ):
+            combine_cmaps("Blues", invalid_cmap)
+
+    # Test if invalid nodes types raise an error
+    @pytest.mark.parametrize("invalid_nodes", [0.5])
```

Comment:
hm, maybe we don't need to parametrize for a single value ?

Reply:
Yes it make sense, thanks for pointing out. In the meantime maybe I should try some packages like `hypothesis` to automatically generate invalid/valid test data here, instead of hard coding.

Reply:
I think hypothesis is overkill for our needs. In general, fewer dependencies makes our code easier to maintain, so let's not introduce stuff we don't need.

Reply:
Yes sure. I might learn this myself just for future proof 😊.